program/rust/src/log.rs

```diff
@@ -2,45 +2,64 @@ use crate::c_oracle_header::*;
 use crate::error::OracleError;
 use borsh::BorshDeserialize;
 use solana_program::account_info::AccountInfo;
+use solana_program::clock::Clock;
 use solana_program::entrypoint::ProgramResult;
 use solana_program::msg;
+use solana_program::sysvar::Sysvar;
 use std::mem::size_of;
 
-pub fn pre_log(accounts : &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
+pub fn pre_log(accounts: &[AccountInfo], instruction_data: &[u8]) -> ProgramResult {
     msg!("Pyth oracle contract");
 
     let instruction_header: cmd_hdr = cmd_hdr::try_from_slice(&instruction_data[..8])?;
     let instruction_id: u32 = instruction_header
         .cmd_
         .try_into()
         .map_err(|_| OracleError::Generic)?;
+
+    let clock = Clock::get()?;
+
+    let expo_start : usize = PRICE_T_EXPO_OFFSET.try_into()
+    .map_err(|_| OracleError::Generic)?;
+
+    let expo: i32 = i32::try_from_slice(
+        &accounts
+            .get(1)
```

Comment:
Could you explain the reason for using index 1 in a comment?

Reply:
Done